src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -55,6 +62,38 @@ internal bool AddGas(long gas)
             return testMode || GasConsumed <= gas_amount;
         }
 
+        internal void CallFromNativeContract(Action onComplete, UInt160 hash, string method, params StackItem[] args)
+        {
+            invocationStates.Add(CurrentContext, new InvocationState
+            {
+                ReturnType = typeof(void),
+                Callback = onComplete
+            });
+            CallContract(hash, method, new VMArray(args));
+        }
+
+        internal void CallFromNativeContract<T>(Action<T> onComplete, UInt160 hash, string method, params StackItem[] args)
+        {
+            invocationStates.Add(CurrentContext, new InvocationState
+            {
+                ReturnType = typeof(T),
+                Callback = onComplete
+            });
+            CallContract(hash, method, new VMArray(args));
+        }
+
+        protected override void ContextUnloaded(ExecutionContext context)
+        {
+            base.ContextUnloaded(context);
+            if (!(UncaughtException is null)) return;
```

Comment:
It's really needed to call from a native contract to a user contract? I think that we should avoid this behaviour